sdk/core/Azure.Core/tests/public/ModelReaderWriterTests/ModelXmlCrossLibraryTests.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.ClientModel;
+using System.Text;
+using System.Text.Json;
+using System.Xml;
+using Azure.Core.Tests.Common;
+using Azure.Core.Tests.Public.ModelReaderWriterTests.Models;
+using NUnit.Framework;
+
+namespace Azure.Core.Tests.Public.ModelReaderWriterTests
+{
+    internal class ModelXmlCrossLibraryTests : ModelJsonTests<ModelXmlCrossLibrary>
+    {
+        protected override string WirePayload => File.ReadAllText(TestData.GetLocation("ModelXmlX.xml")).TrimEnd();
+
+        protected override string JsonPayload => "{\"key\":\"Color\",\"value\":\"Red\",\"readOnlyProperty\":\"ReadOnly\",\"childTag\":{\"childValue\":\"ChildRed\",\"childReadOnlyProperty\":\"ChildReadOnly\"}}";
+
+        [Test]
+        public void ThrowsIfMismatch()
+        {
+            ModelReaderWriterOptions jsonOptions = ModelReaderWriterOptions.Json;
+            ModelXmlCrossLibrary model = ModelReaderWriter.Read<ModelXmlCrossLibrary>(new BinaryData(Encoding.UTF8.GetBytes(JsonPayload)), jsonOptions);
+
+            Assert.Throws(Is.InstanceOf<JsonException>(), () => ModelReaderWriter.Read<ModelXmlCrossLibrary>(new BinaryData(Encoding.UTF8.GetBytes(WirePayload)), jsonOptions));
+
+            ModelReaderWriterOptions wireOptions = ModelReaderWriterHelper.WireOptions;
+            Assert.Throws<XmlException>(() => ModelReaderWriter.Read<ModelXmlCrossLibrary>(new BinaryData(Encoding.UTF8.GetBytes(JsonPayload)), wireOptions));
+        }
+
+        protected override string GetExpectedResult(string format)
+        {
+            if (format == "W")
+            {
+                var expectedSerializedString = "\uFEFF<?xml version=\"1.0\" encoding=\"utf-8\"?><Tag><Key>Color</Key><Value>Red</Value>";
+                if (format.Equals("J"))
+                    expectedSerializedString += "<ReadOnlyProperty>ReadOnly</ReadOnlyProperty>";
+                expectedSerializedString += "<ChildTag><ChildValue>ChildRed</ChildValue></ChildTag>";
+                //TODO this is broken until we update the IXmlSerializable interface to include ModelSerializerOptions
```

Comment:
Are we concerned with leaving inline TODOs?

Reply:
No removed we don't have cross library xml cases so this isn't an issue.